app/overrides/add_label_to_product.rb

```diff
@@ -6,7 +6,7 @@ class AddLabelToProduct
     name: 'add_label_to_product',
     insert_after: "div[data-hook='admin_product_form_tax_category']",
     text: <<-HTML
-    <%- unless @product.labels.blank? %>
+    <%- unless @product.labels&.blank? %>
```

Comment:
Nie do końca rozumiem jak usunięcie safe navigator operatora z tego miejsca ma naprawić błąd `undefined method 'labels' for an instance of Spree::Product`

Jak już chcesz załatać coś januszfixem (bo nawet jak nie ma żadnych labelek, to tu nigdy nie powinien pojawić się błąd undefined method), to zrób to z głową np tak:
```suggestion
    <%- unless @product.try(:labels).blank? %>
```
Ten kod nawet jeśli użyjesz tu metody, która nie istnieje nigdzie zawsze zwróci ci true lub false przy użyciu blank?

Reply:
To prawda co mówi @nika-piotrowska - usunięcie safe navigatora w żaden sposób nie rozwiązuje problemu. 
Użycie `&.` na wywołaniu metody, która nie istnieje, NIE zapobiegnie NoMethodError - safe navigation operator dla przypomnienia podstaw działa tylko jeśli obiekt jest nil, a nie jeśli metoda nie istnieje.

Więc jak dla mnie zmiany w tym miejscu są co całkowitego przywrócenia jak były

Reply:
natrafiłem na błąd, którego szukałem.
problem został rozwiązany za pomocą `@product.try(:labels).blank?`
Dziękuję :)

Reply:
@vitaliicherednichenko przepraszam jaki to problem naprawia? Bo na pewno nie ten o którym jest mowa w opisie. A poza tym, jak widzę z komentarzy wyżej 

> Nie, ten błąd u mnie nie występuje, nawet jeśli usunę wszystkie labelki ze wszystkich sklepów w nowym dumpie.

to nie byłeś w stanie odtworzyć błędu, więc zostało wszystko zaimplementowane w ciemno z nadzieją, że problem zniknie?

Chciałabym, razem z resztą, zrozumieć w jaki sposób ten problem naprawia błąd? 

1. `&.` zabezpiecza tylko przed `nil`-  jeśli metoda nie istnieje i tak rzuci wyjątek
2. `try(:labels)` **ukrywa** błąd "undefined method" i zwraca nil - maskuje prawdziwy problem z asocjacją/interfejsem


więc to nie jest problem w tym miejscu